source/commands/user.cpp

```diff
@@ -1,13 +1,51 @@
 #include "commands.hpp"
 
+#define SPACE ' '
+#define NUL '\0'
+#define CR '\r'
+#define LF '\n'
+
+bool nonwhite(char c)
+{
+	if (c == SPACE)
+		return (false);
+	if (c == NUL)
+		return (false);
+	if (c == CR)
+		return (false);
+	if (c == LF)
+		return (false);
+	return (true);
+}
```

Comment:
Same as above, we might want to put that in an extra file.
But:
Great work! :+1: